tidb-cloud/setup-azure-self-hosted-kafka-private-link-service.md

```diff
@@ -0,0 +1,542 @@
+    {broker-node-3-ip}:9092 (id: 3 rack: null) -> (
+
+    # Bootstrap from EXTERNAL listener
+    ./kafka_2.13-3.7.1/bin/kafka-broker-api-versions.sh --bootstrap-server {one_of_broker_ip}:39092
+    # Expected output for the last 3 lines (the actual order might be different)
+    # The difference in the output from "bootstrap from INTERNAL listener" is that exceptions or errors might occur because advertised listeners cannot be resolved in kafka-pls-vnet.
+    # TiDB Cloud will make them resolvable and make it route to the right broker when you create a changefeed connect to this Kafka cluster by Private Link Service. 
+    b1.abc.eastus.azure.3199745.tidbcloud.com:9093 (id: 1 rack: null) -> ERROR: org.apache.kafka.common.errors.DisconnectException
+    b2.abc.eastus.azure.3199745.tidbcloud.com:9094 (id: 2 rack: null) -> ERROR: org.apache.kafka.common.errors.DisconnectException
+    b3.abc.eastus.azure.3199745.tidbcloud.com.com:9095 (id: 3 rack: null) -> ERROR: org.apache.kafka.common.errors.DisconnectException
+    ```
+
+2. Create a producer script `produce.sh` in the bastion node.
+
+    ```shell
+    BROKER_LIST=$1
+
+    SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+    export JAVA_HOME="$SCRIPT_DIR/jdk-22.0.2"
+    KAFKA_DIR="$SCRIPT_DIR/kafka_2.13-3.7.1/bin"
+    TOPIC="test-topic"
+
+    create_topic() {
+        echo "Creating topic if it does not exist..."
+        $KAFKA_DIR/kafka-topics.sh --create --topic $TOPIC --bootstrap-server $BROKER_LIST --if-not-exists --partitions 3 --replication-factor 3
+    }
+
+    produce_messages() {
+        echo "Producing messages to the topic..."
+        for ((chrono=1; chrono <= 10; chrono++)); do
+            message="Test message "$chrono
+            echo "Create "$message
+            echo $message | $KAFKA_DIR/kafka-console-producer.sh --broker-list $BROKER_LIST --topic $TOPIC
+        done
+    }
+    create_topic
+    produce_messages 
+    ```
+
+3. Create a consumer script `consume.sh` in the bastion node.
+
+    ```shell
+    BROKER_LIST=$1
+    SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+    export JAVA_HOME="$SCRIPT_DIR/jdk-22.0.2"
+    KAFKA_DIR="$SCRIPT_DIR/kafka_2.13-3.7.1/bin"
+    TOPIC="test-topic"
+    CONSUMER_GROUP="test-group"
+    consume_messages() {
+        echo "Consuming messages from the topic..."
+        $KAFKA_DIR/kafka-console-consumer.sh --bootstrap-server $BROKER_LIST --topic $TOPIC --from-beginning --timeout-ms 5000 --consumer-property group.id=$CONSUMER_GROUP
+    }
+    consume_messages
+    ```
+
+4. Execute `produce.sh` and `consume.sh` to verify that the Kafka cluster is running. These scripts will also be reused for later network connection testing. The script will create a topic with `--partitions 3 --replication-factor 3`. Ensure that all these three brokers contain data. Ensure that the script will connect to all three brokers to guarantee that network connection will be tested.
+
```

Comment:
What does "Ensure that the script will connect to all three brokers to guarantee that network connection will be tested." mean?  What can users do to ensure the script will connect to all three brokers?

Reply:
user just execute produce.sh script, produce.sh will help finish the test